packages/did-provider-ebsi/src/EbsiDidProvider.ts

```diff
@@ -17,48 +17,76 @@ export class EbsiDidProvider extends AbstractIdentifierProvider {
-        privateKeyHex = privateKeyHex.substring(2)
-      }
-      if (!privateKeyHex || privateKeyHex.length !== 64) {
-        throw Error('Private key should be 32 bytes / 64 chars hex')
-      }
-      const key = await context.agent.keyManagerImport({
-        type: 'Secp256k1',
-        kms: this.assertedKms(kms),
-        // meta: options?.options?.meta,
-        kid: options?.options?.key?.kid,
-        privateKeyHex,
-      })
-
-      const methodSpecificId = toMethodSpecificId(ebsiDIDSpecInfo.V1, options?.options?.methodSpecificId)
+      const secp256r1ManagedKeyInfo = await this.generateEbsiKeyPair(
+        {
+          keyOpts: {
+            ...options?.secp256r1,
+            type: this.assertedKeyType({ key: options?.secp256r1, keyType: KeyType.Secp256r1 }),
+            purposes: this.assertedPurposes({ keyType: KeyType.Secp256r1, purposes: (options?.secp256r1 as VerificationMethod)?.purposes }),
+          },
+          alias,
+          kms,
+        },
+        context
+      )
+
+      const methodSpecificId = toMethodSpecificId(ebsiDIDSpecInfo.V1, options?.methodSpecificId)
       const identifier: Omit<IIdentifier, 'provider'> = {
         did: ebsiDIDSpecInfo.V1.method + methodSpecificId,
-        controllerKeyId: key.kid,
-        keys: [key],
+        controllerKeyId: secp256k1ManagedKeyInfo.kid,
+        keys: [secp256k1ManagedKeyInfo, secp256r1ManagedKeyInfo],
         services: [],
       }
       debug('Created', identifier.did)
       return identifier
-    } else if (options.type === ebsiDIDSpecInfo.KEY) {
-      throw Error(`Type ${options.type} not supported. Please use @sphereon/ssi-sdk-ext.did-provider-key for Natural Person EBSI DIDs`)
+    } else if (type === ebsiDIDSpecInfo.KEY) {
+      throw Error(`Type ${type} not supported. Please use @sphereon/ssi-sdk-ext.did-provider-key for Natural Person EBSI DIDs`)
+    }
+    throw Error(`Type ${type} not supported`)
+  }
+
+  private async generateEbsiKeyPair(
+    args: { keyOpts?: IKeyOpts | VerificationMethod; alias?: string; kms?: string },
+    context: IAgentContext<IKeyManager>
+  ) {
+    let privateKeyHex = generateEbsiPrivateKeyHex(
+      ebsiDIDSpecInfo.V1,
+      args.keyOpts?.key?.privateKeyHex ? u8a.fromString(args.keyOpts.key.privateKeyHex, 'base16') : undefined
+    )
+    if (privateKeyHex.startsWith('0x')) {
+      privateKeyHex = privateKeyHex.substring(2)
+    }
+    if (!privateKeyHex || privateKeyHex.length !== 64) {
+      throw Error('Private key should be 32 bytes / 64 chars hex')
+    }
+
+    const keyManagerImportArgs: MinimalImportableKey = {
+      type: args.keyOpts?.type!, // It's safe to use '!', a default value is set
+      kms: this.assertedKms(args.kms),
+      kid: args.keyOpts?.kid,
+      privateKeyHex,
     }
-    throw Error(`Type ${options.type} not supported`)
+
+    if (args?.keyOpts && 'purposes' in args.keyOpts) {
+      keyManagerImportArgs.meta = { purposes: [...args.keyOpts.purposes] }
```

Comment:
See remark about purposes. IMO there should always be purposes, as otherwise the keys have no meaning in the context of the DID

Reply:
Refactored the method and added the purpose checks as requested in one of the previous remarks